sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -164,8 +165,11 @@ void gnrc_lorawan_timeout_cb(gnrc_lorawan_t *mac)
         case LORAWAN_STATE_JOIN:
             gnrc_lorawan_trigger_join(mac);
             break;
+        case LORAWAN_STATE_RETRANS:
+            gnrc_lorawan_event_retrans_timeout(mac);
+            break;
         default:
-            gnrc_lorawan_event_ack_timeout(mac);
+            assert(false);
```

Comment:
Assert being triggered while doing tests.

Reply:
As discussed, it seems like tests seem to fail as there are more `mac->state` 's which results in failed assertion.

Reply:
The problem was that the state was immediately overwritten. This assertion is there on purpose, just to make sure there's no wrong state here (and basically it did it's job pretty well)